src/annotator/config/index.js

```diff
@@ -2,52 +2,218 @@ import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
 /**
- * Reads the Hypothesis configuration from the environment.
- *
- * @param {Window} window_ - The Window object to read config from.
+ * @typedef ConfigDefinition
+ * @prop {boolean} allowInBrowserExt - Allow this name to be available in the browser extension
+ * @prop {any} [defaultValue] - Sets a default if `valueFn` returns undefined
+ * @prop {([configName]: string) => any} valueFn -
+ *   Method to retrieve the value from the incoming source
+ * @prop {(value: any) => any} [coerce] - Transform a value's type value or value
  */
-export default function configFrom(window_) {
-  const settings = settingsFrom(window_);
+
+/**
+ * @typedef {Record<string, ConfigDefinition>} ConfigDefinitionMap
```

Comment:
Per suggestion, I changed this to be a Record rather than a defined key: value pair of every setting. This is clearly better.